### PR TITLE
fix(explorer): fix almost filled orders on Explorer

### LINKED ISSUE #4548: Limit: Order does not show '99.99%' of fill
Related to #4482 and #4477 

Link to such order: https://dev.explorer.cow.fi/sepolia/orders/0x7717e411cb91aac3fd4bc787194b9a4e02048e9a361127cb5ef38496dde3c7229fa3c00a92ec5f96b1ad2527ab41b3932efeda5866743a1c?tab=overview

The order is 99.99999999999% filled, but it shows 100% **instead of '>99.99%'** both on CoW Swap and on Explorer. 
The order hangs in "partially filled' state on Exploer, in 'Open' state on CoW Swap' until expires
image
image
image
image

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Fixes #4548

Almost filled orders should show as `>99.99%` instead of `100%` on Explorer

# To Test

1. Check an order that's almost filled (example: `0x0e8e47601b87c3da76d4d2cddd4ddaa18e69a3024c964b94a8fa0497dba9142fc6b13d5e662fa0458f03995bcb824a1934aa895f6682c6c8`)
* Should show `>99.99%` filled instead of `100%`

2. Repeat tests for CoW Swap (https://github.com/cowprotocol/cowswap/pull/4477)
* Should work as before

### CODE REVIEW

File: apps/explorer/src/utils/operator.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export function getOrderFilledAmount(order: RawOrder): { amount: BigNumber; perc
     totalAmount = new BigNumber(buyAmount)
   }
 
-  return { amount: executedAmount, percentage: new BigNumber(executedAmount.toFixed() / totalAmount.toFixed()) }
+  return { amount: executedAmount, percentage: executedAmount.div(totalAmount) }
```
Comment (alfetopito) [inline apps/explorer/src/utils/operator.ts:97]: There where 2 issues.

This is 1, where the amounts were converted from BigNumber to string, divided, then turned into BigNumber.

File: apps/explorer/src/components/orders/FilledProgress/index.tsx [hunk 2]
```diff
@@ -206,7 +206,7 @@ export function FilledProgress(props: Props): JSX.Element {
   const mainSymbol = mainToken ? safeTokenName(mainToken) : mainAddress
   const swappedSymbol = swappedToken ? safeTokenName(swappedToken) : swappedAddress
   // In case the token object is empty, display the raw amount (`decimals || 0` part)
-  const formattedPercentage = filledPercentage.toNumber() * 100
+  const formattedPercentage = filledPercentage.times(100)
```
Comment (alfetopito) [inline apps/explorer/src/components/orders/FilledProgress/index.tsx:209]: Second issue.

BigNumber converted to Number then multiplied by 100, losing the precision from BigNumber.

File: apps/cowswap-frontend/src/utils/orderUtils/parseOrder.ts [hunk 3]
```diff
@@ -26,7 +25,7 @@ export interface ParsedOrderExecutionData {
   surplusPercentage: BigNumber
   executedFeeAmount: string | undefined
   executedSurplusFee: string | null
-  filledPercentDisplay: number
+  filledPercentDisplay: string
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/utils/orderUtils/parseOrder.ts:28]: If it's a `display` amount, why not use string?

File: libs/ui/src/pure/PercentDisplay/index.tsx [hunk 4]
```diff
@@ -1,9 +1,15 @@
-const ALMOST_HUNDRED = 99.99
+import BigNumber from 'bignumber.js'
 
-export const percentIsAlmostHundred = (percent: number) => percent > ALMOST_HUNDRED && percent < 100
+const ALMOST_HUNDRED = BigNumber(99.99)
+
+export function percentIsAlmostHundred(percent: string) {
+  const p = BigNumber(percent)
```
Comment (shoom3301) [inline libs/ui/src/pure/PercentDisplay/index.tsx:6]: Why do we need `bignumber.js` here?
The value is not supposed to be big (>100)?

**Review by elena-zh — APPROVED** [review]
Works, thank you 
image
image

File: libs/ui/src/pure/PercentDisplay/index.tsx (hunk 4 above)
Comment (alfetopito) [reply libs/ui/src/pure/PercentDisplay/index.tsx:6]: The problem is not that's big, but the precision.
For example, this order `0x0e8e47601b87c3da76d4d2cddd4ddaa18e69a3024c964b94a8fa0497dba9142fc6b13d5e662fa0458f03995bcb824a1934aa895f6682c6c8`.

The difference between executed and sell amount is 20 atoms. That's a very small amount that can't be represented with default number precision. 

This is what's shown for this order when using BigNumber

image

And this is after I convert from String to Number with this code
image

image

**Comment by alfetopito** [comment]
Please post merge review

**Pull request merged**